src/analysis/convex/integral.lean

```diff
@@ -4,51 +4,185 @@ Released under Apache 2.0 license as described in the file LICENSE.
+- for convex functions: `convex.on.average_mem_epigraph`, `convex_on.map_average_le`,
+  `convex_on.set_average_mem_epigraph`, `convex_on.map_set_average_le`, `convex_on.map_integral_le`;
 
-* `convex_on.map_integral_le`: Convex Jensen's inequality: If a function `g : E → ℝ` is convex and
-  continuous on a convex closed set `s`, `μ` is a probability measure on `α`, and `f : α → E` is a
-  function sending `μ`-a.e. points to `s`, then the value of `g` at the expected value of `f` is
-  less than or equal to the expected value of `g ∘ f` provided that both `f` and `g ∘ f` are
-  integrable. See also `convex_on.map_sum_le` for a finite sum version of this lemma.
+- for strictly convex sets: `strict_convex.ae_eq_const_or_average_mem_interior`;
+
+- for a closed ball in a strictly convex normed space:
+  `strict_convex.ae_eq_const_or_norm_integral_lt_of_norm_le_const`
+
+- for strictly convex functions: `strict_convex_on.ae_eq_const_or_map_average_lt`.
+
+## TODO
+
+- Add versions for concave functions.
 
 ## Tags
 
-convex, integral, center mass, Jensen's inequality
+convex, integral, center mass, average value, Jensen's inequality
 -/
 
-open measure_theory set filter
-open_locale topological_space big_operators
+open measure_theory measure_theory.measure metric set filter topological_space function
+open_locale topological_space big_operators ennreal convex
 
-variables {α E : Type*} [measurable_space α] {μ : measure α}
+variables {α E F : Type*} {m0 : measurable_space α}
   [normed_group E] [normed_space ℝ E] [complete_space E]
   [topological_space.second_countable_topology E] [measurable_space E] [borel_space E]
+  [normed_group F] [normed_space ℝ F] [complete_space F]
+  [topological_space.second_countable_topology F] [measurable_space F] [borel_space F]
+  {μ : measure α} {s : set E}
+
+/-!
+### Average value of a function w.r.t. a measure
+
+The average value of a function `f` w.r.t. a measure `μ` (notation: `⨍ x, f x ∂μ`) is defined as
+`(μ univ).to_real⁻¹ • ∫ x, f x ∂μ`, so it is equal to zero if `f` is not integrable or if `μ` is an
+infinite measure. If `μ` is a probability measure, then the average of any function is equal to its
+integral.
+
+For the average on a set, use `⨍ x in s, f x ∂μ` (defined as `⨍ x, f x ∂(μ.restrict s)`). For
+average w.r.t. the volume, one can omit `∂volume`.
+-/
+
+namespace measure_theory
+
+variable (μ)
+include m0
+
+/-- Average value of a function `f` w.r.t. a measure `μ`, notation: `⨍ x, f x ∂μ`. It is defined as
+`(μ univ).to_real⁻¹ • ∫ x, f x ∂μ`, so it is equal to zero if `f` is not integrable or if `μ` is an
+infinite measure. If `μ` is a probability measure, then the average of any function is equal to its
+integral.
+
+For the average on a set, use `⨍ x in s, f x ∂μ` (defined as `⨍ x, f x ∂(μ.restrict s)`). For
+average w.r.t. the volume, one can omit `∂volume`. -/
+noncomputable def average (f : α → E) := (μ univ).to_real⁻¹ • ∫ x, f x ∂μ
```

Comment:
What about defining it instead as the integral with respect to the measure `(μ univ).to_real⁻¹ • μ`, to make sure that all integral lemmas (like dominated convergence or whatever) apply directly to this notion? (Just like you did for the integral on a set, defining it as the integral with respect to `μ.restrict s`).